src/super_gradients/training/sg_trainer/sg_trainer.py

```diff
@@ -1562,13 +1563,29 @@ def _initialize_sg_logger_objects(self, additional_configs_to_log: Dict = None):
         if isinstance(sg_logger, AbstractSGLogger):
             self.sg_logger = sg_logger
         elif isinstance(sg_logger, str):
+
+            sg_logger_cls = SG_LOGGERS.get(sg_logger)
+            if sg_logger_cls is None:
+                raise RuntimeError(f"sg_logger={sg_logger} not registered in SuperGradients. Available {list(SG_LOGGERS.keys())}")
+
             sg_logger_params = core_utils.get_param(self.training_params, "sg_logger_params", {})
-            if issubclass(SG_LOGGERS[sg_logger], BaseSGLogger):
+            if issubclass(sg_logger_cls, BaseSGLogger):
                 sg_logger_params = {**sg_logger_params, **general_sg_logger_params}
-            if sg_logger not in SG_LOGGERS:
-                raise RuntimeError("sg_logger not defined in SG_LOGGERS")
 
-            self.sg_logger = SG_LOGGERS[sg_logger](**sg_logger_params)
+            # Some sg_logger require model_name, but not all of them.
+            if "model_name" in get_callable_param_names(sg_logger_cls.__init__):
+                if sg_logger_params.get("model_name") is None:
+                    # Use the model name used in `models.get(...)` if relevant
+                    sg_logger_params["model_name"] = get_model_registration_name(self.net.module)
+
+                if sg_logger_params["model_name"] is None:
+                    raise ValueError(
+                        f'`model_name` is required to use `training_hyperparams.sg_logger="{sg_logger}"`.\n'
+                        'Please set `training_hyperparams.sg_logger_params.model_name="<your-model-name>"`.\n'
+                        "Note that specifying `model_name` is not required when the model was loaded using `models.get(...)`."
+                    )
```

Comment:
I don't like using "model_name" in get_callable_param_names(sg_logger_cls.__init__) but I feel like this is the least dirty solution we can come up with current design.
Another alternative would be to add *args, **kwargs to the SgLoggers but this means that it would silently ignore errors in the usage and/or typos. And adding a kwargs check increases code complexity. So, I think it's actually even more dirty.
If you have an opinion/idea on that I'm Interested